mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java

```diff
@@ -168,6 +168,8 @@ void storeAndUpdateWorkers(final IMantisWorkerMetadata existingWorker, final IMa
 
     void addNewJobArtifactsToCache(ClusterID clusterID, List<ArtifactID> artifacts) throws IOException;
 
+    void removeJobArtifactsToCache(ClusterID clusterID, List<ArtifactID> artifacts) throws IOException;
+
```

Comment:
In order to make it simpler on the storage layer, could we just have a setter instead? what's the need to perform the delta/dedup on both the resource cluster layer as well as the storage layer?

Reply:
I don't know how simpler the implementation with a setter would look like. You'd still need a deleteAll on the pkey (aka clusterID) and addAll on remaining artifacts. I like the consistency of add and remove which will modify both RC and KV store.